server/src/main/java/org/spine3/server/command/CommandHandler.java

```diff
@@ -20,107 +20,66 @@
 
 package org.spine3.server.command;
 
-import com.google.common.collect.ImmutableList;
-import com.google.protobuf.Any;
 import com.google.protobuf.Empty;
 import com.google.protobuf.Message;
-import com.google.protobuf.Timestamp;
 import org.spine3.base.CommandContext;
 import org.spine3.base.Event;
-import org.spine3.base.EventContext;
-import org.spine3.base.EventId;
-import org.spine3.base.FailureThrowable;
-import org.spine3.server.entity.Entity;
 import org.spine3.server.event.EventBus;
-import org.spine3.server.reflect.CommandHandlerMethod;
-import org.spine3.server.reflect.MethodRegistry;
 
-import java.lang.reflect.InvocationTargetException;
 import java.util.List;
 
-import static org.spine3.base.Events.createEvent;
-import static org.spine3.base.Events.generateId;
-import static org.spine3.base.Identifiers.idToAny;
-import static org.spine3.protobuf.Timestamps.getCurrentTime;
-
 /**
  * The abstract base for classes that expose command handling methods
  * and post their results to {@link EventBus}.
  *
  * <p>A command handler is responsible for:
  * <ol>
- *     <li>Changing the state of the business model
+ *     <li>Changing the state of the business model in response to a command.
  *     <li>Producing corresponding events.
  *     <li>Posting events to {@code EventBus}.
  * </ol>
  *
  * <p>Event messages are returned as values of command handling methods.
  *
- * <h2>Command handling methods</h2>
- * <p>A command handling method is a {@code public} method that accepts two parameters.
- * The first parameter is a command message. The second parameter is {@link CommandContext}.
- *
- * <p>The method returns an event message of the specific type, or {@code List} of messages
- * if it produces more than one event.
- *
- * <p>The method may throw one or more throwables derived from {@link FailureThrowable}.
- * Throwing a {@code FailureThrowable} indicates that the passed command cannot be handled
- * because of a business failure, which can be obtained via {@link FailureThrowable#getFailure()}.
+ * <p>A command handler does not have own state. So the state of the business
+ * model it changes is external to it. Even though such a behaviour may be needed in
```

Comment:
There can be cases when it's really needed. We recommend another approach. That should be enough.